build2/KoreBuild.ps1

```diff
@@ -0,0 +1,100 @@
+$MSBuildDir = Join-Path $MSBuildProjDir "publish"
+$KoreBuildTasksPath = Join-Path $ToolsDirectory "KoreBuild"
+
+$KoreBuildLog = Join-Path $PSScriptRoot "..\korebuild.log"
+if(Test-Path $KoreBuildLog) {
+    del $KoreBuildLog
+}
+
+function exec($cmd) {
+    Write-Host -ForegroundColor Cyan "> $cmd $args"
+    "`r`n>>>>> $cmd $args <<<<<`r`n" >> $KoreBuildLog
+    & $cmd @args >> $KoreBuildLog
+    if($LASTEXITCODE -ne 0) {
+        throw "Command returned exit code $($LASTEXITCODE): '$cmd $args'"
+    }
+}
+
+function EnsureDotNet() {
+    $dotnetVersionFile = $PSScriptRoot + "\..\build\cli.version.win"
+    $dotnetChannel = "preview"
+    $dotnetVersion = Get-Content $dotnetVersionFile
+
+    if ($env:KOREBUILD_DOTNET_CHANNEL)
+    {
+        $dotnetChannel = $env:KOREBUILD_DOTNET_CHANNEL
+    }
+    if ($env:KOREBUILD_DOTNET_VERSION)
+    {
+        $dotnetVersion = $env:KOREBUILD_DOTNET_VERSION
+    }
+
+    $dotnetLocalInstallFolder = "$env:LOCALAPPDATA\Microsoft\dotnet\"
+    $newPath = "$dotnetLocalInstallFolder;$env:PATH"
+    if ($env:KOREBUILD_SKIP_RUNTIME_INSTALL -eq "1") 
+    {
+        Write-Host -ForegroundColor Green "Skipping runtime installation because KOREBUILD_SKIP_RUNTIME_INSTALL = 1"
+        # Add to the _end_ of the path in case preferred .NET CLI is not in the default location.
+        $newPath = "$env:PATH;$dotnetLocalInstallFolder"
+    }
+    else
+    {
+        Write-Host -ForegroundColor Green "Installing .NET Command-Line Tools ..."
+        & "$PSScriptRoot\..\build\dotnet\dotnet-install.ps1" -Channel $dotnetChannel -Version $dotnetVersion -Architecture x64 2>&1 >> $KoreBuildLog
+    }
+    if (!($env:Path.Split(';') -icontains $dotnetLocalInstallFolder))
+    {
+        Write-Host -ForegroundColor Green "Adding $dotnetLocalInstallFolder to PATH"
+        $env:Path = "$newPath"
+    }
+
+    # workaround for CLI issue: https://github.com/dotnet/cli/issues/2143
+    $sharedPath = (Join-Path (Split-Path ((get-command dotnet.exe).Path) -Parent) "shared");
+    (Get-ChildItem $sharedPath -Recurse *dotnet.exe) | %{ $_.FullName } | Remove-Item;
+}
+
+function EnsureTools() {
+    if(!(Test-Path $ToolsDirectory)) { mkdir $ToolsDirectory | Out-Null }
+    if(!(Test-Path $MSBuildProjDir)) { mkdir $MSBuildProjDir | Out-Null }
+
+    $content = [IO.File]::ReadAllText((Convert-Path "$PSScriptRoot\project.json.template"))
+    $content = $content.Replace("RUNTIME", $RID)
+
+    [IO.File]::WriteAllText((Join-Path $MSBuildProjDir "project.json"), $content);
+
+    copy "$PSScriptRoot\NuGet.config" "$MSBuildProjDir"
+
+    Write-Host -ForegroundColor Green "Preparing build tools..."
+
+    exec dotnet restore "$MSBuildProjDir\project.json" -v Detailed
+    exec dotnet publish "$MSBuildProjDir\project.json" -o "$MSBuildDir"
+    exec dotnet restore "$PSScriptRoot\.." -v Detailed
+    exec dotnet publish "$PSScriptRoot\..\src\Microsoft.AspNetCore.Build.Tasks" -o "$MSBuildDir" -f "netcoreapp1.0"
```

Comment:
In my other branch where I'm actually building the MSBuild targets, I tried to avoid it but did end up needing custom tasks. Primarily to do things like reading `project.json` to determine if a test project targets `net451`